### PR TITLE
extension: json to go

### PR DESCRIPTION
**Pull request by chyroc**
## Description

Translate JSON into a Go type definition

## Screencast

- command:

image

- success:

image

image

- fail:

image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
Hi @chyroc 👋 

Seems to do the job as expected 🔥 If a hotkey is set, then it's impossible to if something is wrong, I suggest you to use HUD when failures 😄 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by chyroc** [comment]
@pernielsentikaer hi, i fix this pr, please review it again, thanks.

File: extensions/json-to-go/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  "name": "json-to-go",
+  "title": "json-to-go",
+  "description": "Translate JSON into a Go type definition",
+  "icon": "json-to-go.png",
+  "author": "chyroc",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "json-to-go",
+      "title": "JSON to Go type",
```
Comment (pernielsentikaer) [inline extensions/json-to-go/package.json:12]:
```suggestion
      "title": "Convert JSON to Go Type",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @chyroc 👋 

The icon is not ideal to use, it works bad in both dark and white background, can you look into this? Let me know if you need help here 😄 

I suggested some corrections for you to address, we're almost ready 🔥 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/json-to-go/src/json-to-go.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    return;
+  }
+
+  const got = jsonToGo(jsonString, null, null, false);
+  if (got.error) {
+    await showHUD(got.error);
+    return;
+  } else {
+    await parse(got.go);
+    await showHUD("Copied to clipboard");
```
[the remark this answers is not in the document]
Comment (chyroc) [reply extensions/json-to-go/src/json-to-go.ts]: fixed

File: extensions/json-to-go/src/json-to-go.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+export const parse = async (contents: string) => {
+  await Clipboard.copy(contents);
+};
+
+export default async () => {
+  let jsonString = "";
+  try {
+    jsonString = await copy();
+  } catch (e) {
+    await showHUD(`${e}`);
```
[the remark this answers is not in the document]
Comment (chyroc) [reply extensions/json-to-go/src/json-to-go.ts]: fixed

File: extensions/json-to-go/package.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "json-to-go",
+  "title": "json-to-go",
+  "description": "Translate JSON into a Go type definition",
```
[the remark this answers is not in the document]
Comment (chyroc) [reply extensions/json-to-go/package.json]: fixed

**Comment by chyroc** [comment]
@pernielsentikaer hi, i fix this pr, please review it again, thanks.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Awesome @chyroc 🔥 

Could you enlarge the icon a bit, is that possible? 😄

**Comment by chyroc** [comment]
@pernielsentikaer hi, i fix this pr, please review it again, thanks.

**Review by pernielsentikaer — APPROVED** [review]
Hi @chyroc 🔥 

Much better (but liked the old icon better 😄 )

Accepted, let's await final review where small typos can be suggested 😄

**Comment by chyroc** [comment]
thanks.

**Pull request merged**